agent/tests/test_lib_aws_sdk_php.c

```diff
@@ -483,20 +554,397 @@ static void test_nr_lib_aws_sdk_php_handle_version(void) {
 }
 
 #if ZEND_MODULE_API_NO >= ZEND_8_1_X_API_NO
+static void setup_inherited_classes() {
+  tlib_php_request_eval(
+      "class endpoint_class{"
+      "public ?string $host;"
+      "public ?int $port;"
+      "function __construct(?int $port = null, ?string $host = null) {"
+      "$this->host = $host;"
+      "$this->port = $port;"
+      "}"
+      "}"
+      "class base_class {"
+      "private ?string $region;"
+      "private ?endpoint_class $endpoint;"
+      "function base_func($command, $args, $expects) {return;}"
+      "function __construct(?string $region = null, ?int $port = null, ?string "
+      "$host = null) {"
+      "$this->region = $region;"
+      "$this->endpoint = new endpoint_class($port, $host);"
+      "}"
+      "}"
+      "class top_class extends base_class {"
+      "}");
```

Comment:
suggestion to make this construct a little more human readable:
```suggestion
  // clang-format off
  const char* endpoint_class = 
      "class endpoint_class {"
        "public ?string $host;"
        "public ?int $port;"
        "function __construct(?int $port = null, ?string $host = null) {"
          "$this->host = $host;"
          "$this->port = $port;"
        "}"
      "}"
      "class base_class {"
        "private ?string $region;"
        "private ?endpoint_class $endpoint;"
        "function base_func($command, $args, $expects) {return;}"
        "function __construct(?string $region = null, ?int $port = null, ?string $host = null) {"
          "$this->region = $region;"
          "$this->endpoint = new endpoint_class($port, $host);"
        "}"
      "}"
      "class top_class extends base_class {}";
  // clang-format on
  tlib_php_request_eval(endpoint_class);
```